source/core/src/main/com/csse3200/game/services/DialogueBoxService.java

```diff
@@ -56,11 +58,32 @@ public Boolean getIsVisible() {
         return currentOverlay.getIsVisible();
     }
 
+    /**
```

Comment:
This seems like an extremely overcomplicated way to do this - is it really necessary to check the call stack? Can you not just create an enum in here which map to priority values (ie call updateText(String[] text, DialoguePriorityEnum))? The current changes seems to increase coupling - if anyone changes the name of any of these classes they'll need to remember to change it here as well.

Reply:
By DialoguePriorityEnum, you mean like an integer value right?

Reply:
Yes

Reply:
Have re-reviewed - only one small change requested now.